pkg/controller/migmigration/registry.go

```diff
@@ -539,6 +539,10 @@ func (t *Task) equalsRegistryService(a, b *kapi.Service) bool {
 // Returns the right backup/restore annotations including registry-specific ones
 func (t *Task) getAnnotations(registryResources *MigRegistryResources) (map[string]string, error) {
 	annotations := t.Annotations
+	// registryResources will only be nil when reconcile is called after restore is complete
+	if registryResources == nil {
+		return annotations, nil
+	}
 	if len(registryResources.Service.Spec.Ports) == 0 {
```

Comment:
I know this isn't part of the current PR, but curious if it's safe to assume that `registryResources.Service.Spec` will exist here... Any possibility this will eval to nil and crash the controller?

Reply:
Perhaps a check for existence of `registryResources.Service != nil` would be enough to prevent this.

Reply:
So the way the code is now, if the registry ensure methods have run, then we will have `registryResources.Service.Spec`. I had to add the nil check because if reconcile is restarted after registry deletion (and after setting `skipMigRegistry`), then registryResources will be nil. It might be safer to check for nil here too, but, in general, it seems that we're only checking for nil in cases where there's code that allows it to be nil.

Reply:
But yeah, I guess I could replace `if registryResources == nil` with `if registryResources == nil || registryResources.Service == nil` and just return without adding a registry hostname annotation if there is no Service here.